lib/OBSWebSocket.js

```diff
@@ -20,50 +20,51 @@ class OBSWebSocket extends Socket {
   // Generates a messageId internally and will override any passed in the args.
   // Note that the requestType here is pre-marshaling and currently must match exactly what the websocket plugin is expecting.
   send(requestType, args = {}, callback) {
+    // TODO: Improve this to ensure args an object, not a function or primitive or something.
     args = args || {};
 
     return new Promise((resolve, reject) => {
-      if (!this._connected) {
-        debug('[send] %s', Status.NOT_CONNECTED.description);
-        this._doCallback(callback, Status.NOT_CONNECTED, null);
-        reject(Status.NOT_CONNECTED);
-        return;
-      }
+      const messageId = generateMessageId();
+      let rejectReason;
 
       if (!requestType) {
-        debug('[send] %s', Status.REQUEST_TYPE_NOT_SPECIFIED.description);
-        this._doCallback(callback, Status.REQUEST_TYPE_NOT_SPECIFIED, null);
-        reject(Status.REQUEST_TYPE_NOT_SPECIFIED);
-        return;
+        rejectReason = Status.REQUEST_TYPE_NOT_SPECIFIED;
       }
 
-      // Assign the core message details.
-      args['request-type'] = requestType;
-      const messageId = args['message-id'] = generateMessageId(); // eslint-disable-line no-multi-assign
-
-      // Submit the request to the websocket.
-      debug('[send] %s %s %o', messageId, requestType, args);
-      try {
-        this._socket.send(JSON.stringify(args));
-      } catch (e) {
-        reject(e);
+      if (!this._connected) {
+        rejectReason = Status.NOT_CONNECTED;
       }
 
       // Assign a temporary event listener for this particular messageId to uniquely identify the response.
       this.once('obs:internal:message:id-' + messageId, message => {
-        // TODO: Do additional stuff with the msg to determine errors, marshaling, etc.
-        // message = API.marshalResponse(requestType, message);
-
         if (message.status === 'error') {
           debug('[send:reject] %o', message);
-          this._doCallback(callback, message, null);
+          this._doCallback(callback, message);
           reject(message);
         } else {
           debug('[send:resolve] %o', message);
           this._doCallback(callback, null, message);
           resolve(message);
         }
       });
+
+      if (!rejectReason) {
+        args['request-type'] = requestType;
+        args['message-id'] = messageId;
+
+        // Submit the request to the websocket.
+        debug('[send] %s %s %o', messageId, requestType, args);
+        try {
+          this._socket.send(JSON.stringify(args));
+        } catch (e) {
+          // TODO: Consider inspecting the exception thrown to gleam some relevant info and pass that on.
+          rejectReason = Status.SOCKET_EXCEPTION;
+        }
+      }
+
+      if (rejectReason) {
+        this.emit('obs:internal:message:id-' + messageId, rejectReason);
```

Comment:
Will likely wrap these emits in a function for consistency. `emitIinternal()` or something.